### PR TITLE
Dataparsers for ScanNet and ARKitScenes

### PR DESCRIPTION
**Pull request by ponimatkin**
This PR adds dataparsers that allow working with ScanNet and ARKitScenes datasets out of box.

Bellow I attach results obtained by running `ns-train nerfacto arkit-data`:

https://user-images.githubusercontent.com/38923619/221418322-f5f64f15-12f8-4188-a36b-2ea75a2fe960.mp4

And results obtained by running `ns-train nerfacto scannet-data`:

https://user-images.githubusercontent.com/38923619/221418342-6c778d29-8333-455d-b4a3-b97eefc76ee7.mp4

### CODE REVIEW

File: nerfstudio/data/dataparsers/arkitscenes_dataparser.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+from nerfstudio.data.dataparsers.base_dataparser import (
+    DataParser,
+    DataParserConfig,
+    DataparserOutputs,
+)
+from nerfstudio.data.scene_box import SceneBox
+
+
+# Taken from https://github.com/apple/ARKitScenes/blob/main/threedod/benchmark_scripts/utils/tenFpsDataLoader.py
+def traj_strin_to_matrix(traj_str: str):
```
Comment (tancik) [inline nerfstudio/data/dataparsers/arkitscenes_dataparser.py:35]: `strin` -> `string`

**Review by tancik — COMMENTED** [review]
Great to see new dataparsers being added! Doesn't need to be in this PR, but it would be convenient to add scannet and arkit to the `ns-download-data` script.

Running `ns-dev-test` will show you where the pylint errors are.

**Comment by ponimatkin** [comment]
Commit e3d08c576832bfef201ededc1b85d392c60ba591 seems to have addressed the pylint errors. Do you think it si good to go now?

As for adding datasets to `ns-download-data`, I believe it might be problematic for ScanNet as you have to sign license agreement to download it. For ARKitScenes, it might be possible, but it is better to ask in ARKitScenes repo for this as I'm not well versed in licensing...

File: nerfstudio/data/dataparsers/arkitscenes_dataparser.py [hunk 2]
```diff
@@ -32,10 +32,10 @@
 
 
 # Taken from https://github.com/apple/ARKitScenes/blob/main/threedod/benchmark_scripts/utils/tenFpsDataLoader.py
-def traj_strin_to_matrix(traj_str: str):
-    """convert traj_str into translation and rotation matrices
+def traj_strin_to_matrix(traj_string: str):
```
Comment (tancik) [inline nerfstudio/data/dataparsers/arkitscenes_dataparser.py:35]: Ahh, I was referring to the typo in the function name.

Comment (ponimatkin) [reply nerfstudio/data/dataparsers/arkitscenes_dataparser.py:35]: Ok, let me quickly fix!

**Comment by tancik** [comment]
Good points regarding downloading the data.

File: nerfstudio/data/dataparsers/arkitscenes_dataparser.py (hunk 2 above)
Comment (ponimatkin) [reply nerfstudio/data/dataparsers/arkitscenes_dataparser.py]: Ok, fixed in e279656454a7d5be5c340ca9e0eac111309b9fed.

**Review by tancik — APPROVED** [review]
LGTM!

**Pull request merged**